x/gov/keeper/keeper.go

```diff
@@ -32,8 +33,12 @@ type Keeper struct {
 	// The codec codec for binary encoding/decoding.
 	cdc codec.BinaryCodec
 
-	// Proposal router
-	router types.Router
+	// Proposal router (this is used for Message based proposals - V2)
+	router *middleware.MsgServiceRouter
+
+	// Proposal Content Handler (this is used for Content based proposals)
+	// NOTE: This is deprecated and will be removed in the next release
+	handler types.Router
```

Comment:
what about keeping the old `types.Router` as `router` and use `msgServiceRouter` for `*middleware.MsgServiceRouter`?